block/deserialize.go

```diff
@@ -0,0 +1,341 @@
+package block
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/vulpemventures/go-elements/internal/bufferutil"
+
+	"github.com/vulpemventures/go-elements/transaction"
+)
+
+func deserialize(buf *bytes.Buffer) (*Block, error) {
```

Comment:
why this function is unexposed?

Reply:
Since I have NewFromHex and NewFromBuffer, which is only invoking deserialise

Reply:
Ah ok, but why other functions here have been made public then?

Reply:
I made public DeserializeHeader and DeserializeTransactions, thought i might be useful  ...